maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/main/java/org/apache/maven/scm/provider/git/jgit/command/JGitTransportConfigCallback.java

```diff
@@ -19,47 +19,22 @@
  * under the License.
  */
 
-import com.jcraft.jsch.JSch;
-import com.jcraft.jsch.JSchException;
-import com.jcraft.jsch.Session;
-import org.apache.maven.scm.provider.git.repository.GitScmProviderRepository;
 import org.eclipse.jgit.api.TransportConfigCallback;
-import org.eclipse.jgit.transport.JschConfigSessionFactory;
-import org.eclipse.jgit.transport.OpenSshConfig;
-import org.eclipse.jgit.transport.SshSessionFactory;
 import org.eclipse.jgit.transport.SshTransport;
 import org.eclipse.jgit.transport.Transport;
-import org.eclipse.jgit.util.FS;
-import org.eclipse.jgit.util.StringUtils;
-import org.slf4j.Logger;
+import org.eclipse.jgit.transport.sshd.SshdSessionFactory;
 
 /**
  * Implementation of {@link TransportConfigCallback} which adds
  * a public/private key identity to ssh URLs if configured.
  */
 public class JGitTransportConfigCallback implements TransportConfigCallback
 {
-    private SshSessionFactory sshSessionFactory = null;
+    private final SshdSessionFactory sshSessionFactory;
 
-    public JGitTransportConfigCallback( GitScmProviderRepository repo, Logger logger )
+    public JGitTransportConfigCallback( SshdSessionFactory sshSessionFactory )
     {
-        if ( repo.getFetchInfo().getProtocol().equals( "ssh" ) )
```

Comment:
Is the value normalized to lowercase with the get?

Reply:
This code line is removed at all